devel/apache-arrow/Portfile

```
@@ -0,0 +1,223 @@
+
+checksums           rmd160  26a1438b995ca7c3fc8ff65d703bfb29133f5ac1 \
+                    sha256  56cc21a24d68a24ccdff043b4a4cec57206a3be5e5b128608c81dbc20dd843ce \
+                    size    6847581
+
+compiler.cxx_standard   2011
+
+universal_variant   no
+
+# Build instructions: https://github.com/apache/arrow/blob/master/docs/source/developers/python.rst#build-and-test
+cmake.source_dir    ${worksrcpath}/cpp
+cmake.build_type    Release
+
+set llvm_version    10
+set llvm_prefix     ${prefix}/libexec/llvm-${llvm_version}
+
+cmake.module_path-append \
+                    ${llvm_prefix}/lib/cmake/llvm
+
+depends_build-append \
+                    port:pkgconfig
+
+depends_lib-append \
+                    port:llvm-${llvm_version}
+
+pre-configure {
+    xinstall -d ${workpath}/bin
+    foreach f [glob -nocomplain ${prefix}/bin/ll*-${llvm_version}] {
+        ln -s ${f} ${workpath}/bin/[strsed [file tail ${f}] "s|-${llvm_version}$||"]
+    }
+}
+
+configure.args-append \
+                    -DARROW_DEPENDENCY_USE_SHARED:BOOL=ON \
+                    -DARROW_FLIGHT:BOOL=ON \
+                    -DARROW_JEMALLOC:BOOL=ON \
+                    -DARROW_ORC:BOOL=ON \
+                    -DARROW_PARQUET:BOOL=ON \
+                    -DARROW_PLASMA:BOOL=ON \
+                    -DARROW_PROTOBUF_USE_SHARED=:BOOLON \
+                    -DARROW_PYTHON:BOOL=ON \
+                    -DARROW_USE_CCACHE:BOOL=OFF \
+                    -DARROW_WITH_BZ2:BOOL=ON \
+                    -DARROW_WITH_ZLIB:BOOL=ON \
+                    -DARROW_WITH_ZSTD:BOOL=ON \
+                    -DARROW_WITH_LZ4:BOOL=ON \
+                    -DARROW_WITH_SNAPPY:BOOL=ON \
+                    -DARROW_WITH_BROTLI:BOOL=ON \
+                    -DARROW_INSTALL_NAME_RPATH:BOOL=OFF \
+                    -DgRPC_ROOT=${prefix} \
+                    -DLLVM_ROOT=${llvm_prefix}
+
+# used for both apache-arrow and py*-pyarrow
+destroot.target     install
+
+if {${name} eq ${subport}} {
+    # Apache Arrow requires a python3 version to build; use this
+    set python_branch   3.8
+    set python_version  [join [split ${python_branch} .] ""]
+    set python_prefix   ${frameworks_dir}/Python.framework/Versions/${python_branch}
+    set python_bin      ${python_prefix}/bin/python${python_branch}
+
+    depends_build-append \
+                    port:python${python_version} \
+                    port:py${python_version}-cython \
+                    port:py${python_version}-numpy
+
+    depends_lib-append \
+                    port:boost \
+                    port:brotli \
+                    port:bzip2 \
+                    port:c-ares \
+                    port:flatbuffers \
+                    port:gmake \
+                    port:grpc \
+                    port:google-glog \
+                    port:jemalloc \
+                    port:libunwind \
+                    port:lz4 \
+                    port:ninja \
+                    port:openssl \
```

If you know for certain that this port is incompatible with libressl, please add a comment with a URL to the upstream bug report about that.

If you don't know for certain that it's incompatible with libressl, allow libressl to be used by declaring this as a path dependency.

```suggestion
                    path:lib/libssl.dylib:openssl \
```

---

Thanks done.